packages/colorpickers/stories/examples/ColorDialog/WithIconButton.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import { Story, Meta } from '@storybook/react';
+import { ToggleIconButton } from '@zendeskgarden/react-buttons';
+import { ColorDialog, IColor } from '@zendeskgarden/react-colorpickers';
+import LeafIcon from '@zendeskgarden/svg-icons/src/16/leaf-fill.svg';
+
+export default {
+  title: 'Components/ColorDialog',
+  component: ColorDialog
+} as Meta;
+
+export const WithIconButton: Story = ({
+  placement,
+  alphaSlider,
+  hueSlider,
+  hex,
+  red,
+  green,
+  blue,
+  alpha,
+  disabled,
+  hasArrow,
+  isAnimated
+}) => {
+  const labels = { alphaSlider, hueSlider, hex, red, green, blue, alpha };
+  const [isPressed, setIsPressed] = useState(false);
+  const [color, setColor] = useState<string | IColor>('rgba(23, 73, 77, 100)');
+  const [selectedColor, setSelectedColor] = useState<string | IColor>('rgba(23, 73, 77, 100)');
+  const iconColor =
+    typeof selectedColor === 'string'
+      ? selectedColor
+      : `rgba(${selectedColor.red}, ${selectedColor.green}, ${selectedColor.blue}, ${selectedColor.alpha})`;
+
+  return (
+    <div style={{ display: 'flex', justifyContent: 'center' }}>
+      <ColorDialog
+        color={color}
+        labels={labels}
+        onChange={setColor}
+        hasArrow={hasArrow}
+        placement={placement}
+        isAnimated={isAnimated}
+        onClick={() => setIsPressed(true)}
+        onClose={nextColor => {
+          setIsPressed(false);
+          setSelectedColor(nextColor);
+        }}
+      >
+        <ToggleIconButton
+          aria-label="leaf"
+          disabled={disabled}
+          isPressed={isPressed}
+          style={{ color: disabled ? undefined : iconColor }}
+        >
+          <LeafIcon />
```

Comment:
The new palette icon that shipped today with `svg-icons` v6.29.0 would be preferable here.